### PR TITLE
hello_world sample

### PR DESCRIPTION
**Pull request by sojinkim-icon**
- sample contract
- README.md
- test

### CODE REVIEW

File: hello_world/README.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    @external(readonly=True)
+    def hello(self) -> str:
+        Logger.info(f'Hello, world!', TAG)
+        return "Hello"
+
+    @payable
+    def fallback(self) -> None:
+        Logger.info(f'fallback is called', TAG)
+        pass
```
Comment (sink772) [inline hello_world/README.md]: Don't need `pass` here.

File: hello_world/README.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    @external
+    def tokenFallback(self, _from: Address, _value: int, _data: bytes) -> None:
+        Logger.info(f'tokenFallabck is called', TAG)
+        pass
```
Comment (sink772) [inline hello_world/README.md]: Don't need `pass` here, too.

File: hello_world/README.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+
+
+## GitHub
+
+Please refer to [Github](https://github.com/icon-project/samples/) for complete source code.
+
+### Invoke a method from CLI
+
+Below is the T-Bears CLI comand to invoke the SCORE's external read-only method. Before issuing the command, don't forget to change the "to" value in the JSON request with an actual SCORE address.
```
Comment (sink772) [inline hello_world/README.md]: typo: comand -> command

File: hello_world/hello_world.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    @external(readonly=True)
+    def hello(self) -> str:
+        Logger.info(f'Hello, world!', TAG)
+        return "Hello"
+
+    @payable
+    def fallback(self) -> None:
+        Logger.info(f'fallback is called', TAG)
+        pass
```
Comment (sink772) [inline hello_world/hello_world.py]: remove `pass`

File: hello_world/hello_world.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    @external
+    def tokenFallback(self, _from: Address, _value: int, _data: bytes) -> None:
+        Logger.info(f'tokenFallabck is called', TAG)
+        pass
```
Comment (sink772) [inline hello_world/hello_world.py]: remove `pass`

**Review by sink772 — CHANGES_REQUESTED** [review]

File: hello_world/README.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    @external
+    def tokenFallback(self, _from: Address, _value: int, _data: bytes):
+        Logger.info('tokenFallabck is called', TAG)
+```
+
+`fallback` function is added to accept ICX. `fallback` function is executed when the contract receives a transaction request without `data` part. Not having `data` in the transaction means no method name is specified. In such case, `fallback` function is invoked if the function is provided. If `fallback` function is not given, the transaction will fail. 
+
+ICX transfer request message does not have `data` part. So, the default behavior of any contract is rejecting incoming ICX. This design is to prevent any accidental ICX transfer to a contract. 
+
+Only functions with `@payable` decorator are permitted to receive ICX coins, therefore, our `fallback` function should be also  `@payable` because we want the contract receive ICX.  You don't need to implement anything in the fallback function to handle the ICX transfer. Having payable fallback function declares that the contract is designed to accept ICX coin transfer.
```
Comment (Spl3en) [inline hello_world/README.md]:
```
--- we want the contract receive ICX
+++ we want the contract to receive ICX
```


```
--- You don't need to implement anything in the fallback function to handle the ICX transfer. Having payable fallback function declares that the contract is designed to accept ICX coin transfer. 
+++ By design, `@payable` functions handle ICX coin transfers, so you don't need to implement anything regarding this matter for these functions.

Comment (Spl3en) [inline hello_world/README.md]:
```
--- we want the contract receive ICX
+++ we want the contract to receive ICX
```


```
--- You don't need to implement anything in the fallback function to handle the ICX transfer. Having payable fallback function declares that the contract is designed to accept ICX coin transfer. 
+++ By design, `@payable` functions handle ICX coin transfers, so you don't need to implement anything regarding this matter for these functions.

File: hello_world/README.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+ICX transfer request message does not have `data` part. So, the default behavior of any contract is rejecting incoming ICX. This design is to prevent any accidental ICX transfer to a contract. 
+
+Only functions with `@payable` decorator are permitted to receive ICX coins, therefore, our `fallback` function should be also  `@payable` because we want the contract receive ICX.  You don't need to implement anything in the fallback function to handle the ICX transfer. Having payable fallback function declares that the contract is designed to accept ICX coin transfer. 
+
+`tokenFallback` method is added to accept IRC2 tokens. The [IRC2](https://github.com/icon-project/IIPs/blob/master/IIPS/iip-2.md#token-fallback) standard mandates the `tokenFallback` method in the receiving SCORE.  Token contact will call the `tokenFallback` funtion whenever it transfers  token to a contract address.
```
Comment (Spl3en) [inline hello_world/README.md]:
```
--- Token contact 
+++ Token contract
```

```
--- funtion
+++ function
```

**Review by Spl3en — CHANGES_REQUESTED** [review]

File: hello_world/README.md (hunk 6 above)
Comment (sojinkim-icon) [reply hello_world/README.md]: fixed the type. 
for the "@payable", it is not exactly the payable function who handles the coin transfer.  iconservice handles the coin transfer, existence of payable fallback function confirms the contract is valid recipient.

**Review by sink772 — APPROVED** [review]

**Pull request merged**